synthesizer/src/program/instruction/operation/cast.rs

```diff
@@ -526,6 +445,60 @@ impl<N: Network> Cast<N> {
     }
 }
 
+impl<N: Network> Cast<N> {
+    /// A helper method to handle casting to a struct.
+    fn cast_to_struct(
+        &self,
+        stack: &Stack<N>,
+        registers: &mut impl Store<N>,
+        struct_name: Identifier<N>,
+        inputs: Vec<Value<N>>,
+    ) -> Result<()> {
+        // Ensure the operands length is at least the minimum.
+        if inputs.len() < N::MIN_STRUCT_ENTRIES {
+            bail!("Casting to a struct requires at least {} operand", N::MIN_STRUCT_ENTRIES)
+        }
+
+        // Retrieve the struct and ensure it is defined in the program.
+        let struct_ = stack.program().get_struct(&struct_name)?;
+
+        // Ensure that the number of operands is equal to the number of struct members.
+        if inputs.len() != struct_.members().len() {
+            bail!(
+                "Casting to the struct {} requires {} operands, but {} were provided",
+                struct_.name(),
+                struct_.members().len(),
+                inputs.len()
+            )
+        }
+
+        // Initialize the struct members.
+        let mut members = IndexMap::new();
+        for (member, (member_name, member_type)) in inputs.iter().zip_eq(struct_.members()) {
+            // Compute the register type.
+            let register_type = RegisterType::Plaintext(*member_type);
+            // Retrieve the plaintext value from the entry.
+            let plaintext = match member {
+                Value::Plaintext(plaintext) => {
+                    // Ensure the member matches the register type.
+                    stack.matches_register_type(&Value::Plaintext(plaintext.clone()), &register_type)?;
```

Comment:
it's not introduced by this PR, but we can avoid this clone by using the publicly available `Stack::matches_plaintext`